applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_geo_steady_state_piping_element.cpp

```diff
@@ -381,4 +381,124 @@ KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementReturnsEquilibriumHeightF
+
+    // Assert
+    KRATOS_EXPECT_DOUBLE_EQ(pipe_heights[0], pipe_height);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementReturnsPermeabilityMatrix, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto dummy_process_info = ProcessInfo{};
+    const auto p_properties       = std::make_shared<Properties>();
+
+    Model model;
+    auto& r_model_part = CreateModelPartWithWaterPressureVariableAndVolumeAcceleration(model);
+    auto  p_element =
+        CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithPWDofs(r_model_part, p_properties);
+
+    const double pipe_height = 1.E-1;
+    p_element->SetValue(PIPE_HEIGHT, pipe_height);
+
+    std::vector<Matrix> permeability_matrices(
+        p_element->GetGeometry().IntegrationPointsNumber(p_element->GetIntegrationMethod()));
+    p_element->CalculateOnIntegrationPoints(PERMEABILITY_MATRIX, permeability_matrices, dummy_process_info);
+
+    // Assert
+    KRATOS_EXPECT_DOUBLE_EQ(permeability_matrices[0](0, 0), std::pow(pipe_height, 3) / 12.0);
+
+    p_element->CalculateOnIntegrationPoints(LOCAL_PERMEABILITY_MATRIX, permeability_matrices, dummy_process_info);
+
+    // Assert
+    KRATOS_EXPECT_DOUBLE_EQ(permeability_matrices[0](0, 0), std::pow(pipe_height, 3) / 12.0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementReturnsFluidFluxVector, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto dummy_process_info = ProcessInfo{};
+    const auto p_properties       = std::make_shared<Properties>();
+
+    Model model;
+    auto& r_model_part = CreateModelPartWithWaterPressureVariableAndVolumeAcceleration(model);
+    auto  p_element =
+        CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithPWDofs(r_model_part, p_properties);
+    p_element->GetProperties().SetValue(DENSITY_WATER, 1.0E3);
+    p_element->GetProperties().SetValue(DYNAMIC_VISCOSITY, 1.0E-2);
+    p_element->GetProperties().SetValue(PIPE_HEIGHT, 1.0E-1);
+    p_element->GetProperties().SetValue(DENSITY_SOLID, 2.0E3);
+    // PIPE_ETA such that multiplication with Pi/3 becomes 1.
+    p_element->GetProperties().SetValue(PIPE_ETA, 3.0 / Globals::Pi);
+    p_element->GetProperties().SetValue(PIPE_MODEL_FACTOR, 1.0);
+    // slope = 0. PIPE_THETA = 45 deg. such that sin(theta + slope) / cos(theta) becomes 1.
+    p_element->GetProperties().SetValue(PIPE_THETA, 45.0);
+    // no PIPE_MODIFIED_D so the equilibrium height becomes PIPE_D_70 / |dh/dx| = 7.e-3 / |-1.e-3| = 7.
+    p_element->GetProperties().SetValue(PIPE_D_70, 7.e-3);
+
+    // Set gravity perpendicular to the line ( so no fluid body flow vector from this )
+    p_element->GetGeometry()[0].FastGetSolutionStepValue(VOLUME_ACCELERATION) =
+        array_1d<double, 3>{0.0, -10.0, 0.0};
+    p_element->GetGeometry()[1].FastGetSolutionStepValue(VOLUME_ACCELERATION) =
+        array_1d<double, 3>{0.0, -10.0, 0.0};
```

Comment:
Perhaps giving the vector a name avoids duplication and improves readability...?
```suggestion
    const auto gravity_acceleration = array_1d<double, 3>{0.0, -10.0, 0.0};
    p_element->GetGeometry()[0].FastGetSolutionStepValue(VOLUME_ACCELERATION) =
        gravity_acceleration;
    p_element->GetGeometry()[1].FastGetSolutionStepValue(VOLUME_ACCELERATION) =
        gravity_acceleration;
```

Reply:
done ( also on the earlier occasions. )